cognite/powerops/resync/v1/config_to_fdm.py

```diff
@@ -0,0 +1,528 @@
+                reference_data[key] = value
+            elif isinstance(value, list):
+                immutable_subdata = []
+                reference_subdata = []
+                for item in value:
+                    if isinstance(item, str) and (item.startswith("[external_id") or item.startswith("[name")):
+                        reference_subdata.append(item)
+                    else:
+                        immutable_subdata.append(item)
+                if reference_subdata:
+                    reference_data[key] = reference_subdata
+                if immutable_subdata:
+                    immutable_data[key] = immutable_subdata
+            else:
+                immutable_data[key] = value
+
+        return immutable_data, property_configuration_data, reference_data
+
+    @staticmethod
+    def _join_data_by_parsing_method(
+        immutable_data: dict, property_configuration_data: dict, reference_data: dict
+    ) -> dict:
+        """Joins dictionaries together in a prioritized order based on the parsing method.
+
+        Provided the data input the dictionaries will be joined together in a prioritized order based on the parsing
+        method. The immutable data will override any other data, followed by the reference data and lastly the
+        property configuration data so that any general property configurations will be overridden by the specific
+        instance data.
+
+        Args:
+            immutable_data: A dictionary of data that will not be overridden by any other settings.
+            property_configuration_data: A dictionary of data that will be overridden from other sources.
+            reference_data: A dictionary of data that will be overridden or joined only with the immutable data.
+
+        Returns:
+            joined_data: A dictionary of all data joined together in a prioritized order based on the parsing method.
+        """
+
+        joined_data = property_configuration_data.copy()
+        joined_data.update(reference_data)
+
+        for key, value in immutable_data.items():
+            if key in joined_data and isinstance(joined_data[key], list):
+                joined_data[key].extend(value)
+            else:
+                joined_data[key] = value
+
+        return joined_data
+
+    def _populate_properties_from_configurations(
+        self, property_configuration_data: dict, property_configurations: list[PropertyConfiguration], raw_data: dict
+    ) -> tuple[dict[str, Any], dict[str, v1_data_classes.DomainModelWrite]]:
+        """Populates properties from configurations based on the provided data.
+
+        Provided the data input and property configurations the data will be parsed based on the property configurations
+        and the data will be updated with the parsed properties and any generated data model objects. The property
+        configurations will be prioritized based on the union of the default property configurations and the override
```

Comment:
What does "_prioritized_ based on the union of..." mean?

Reply:
Any fields configured directly on the instance property have priority over the configuration provided for properties of that type. So it's a union of all the property configurations from the "default" and for the instance, with the instance configuration taking priority.